### ISSUE TITLE
Scope limit secrets. Allow some secrets to be available on build time

### ISSUE THREAD

**Issue by Maks3w** [docker/compose#7936]
**Is your feature request related to a problem? Please describe.**
Docker secrets support can be used for runtime secrets (like database passwords) and also while building the container (pull dependencies from private repositories)

Actually all the secrets defined in the composer service are available while the container is running.

The proposal is allow to specify when the secret should be deployed with the container or only available when building the container.

The actual way is to build the service in a step apart from the rest of services declared in the compose file. The downsides are:

* Lack of parallelism (specific containers must to be build apart from the rest before or after)
* Extra CLI steps & commands (require the use of build scripts)
* Manage a list of exceptions when building some containers with `docker-compose build` vs containers built with `docker build`

**Describe the solution you'd like**
An option it could be add a config key in the service secret long syntax. Like

`service.<service name>.secrets.<secret name>.scope = all/build`

Ex:

```yml
version: "3.8"
services:
  redis:
    image: redis:latest
    deploy:
      replicas: 1
    secrets:
      - source: ssh_key
        scope: build
secrets:
  ssh_key:
    file: ./ssh_key
```

By default it must have the `all` value for backward compatibility.

**Describe alternatives you've considered**
There is an alternative proposed in https://github.com/docker/compose/pull/7046 where the "secrets" service key can be placed as a subkey of service build key (`service.<service name>.build.secrets: -<secret name>`)

However I think is not a good approach to spawn secrets as subkeys as they make difficult to audit and to learn about the specifics. To have a config key in the parent service secrets key make easier to documentate and also make people aware about the risks about not limiting the scope.

**Additional context**
This is related with other features like https://github.com/docker/compose/issues/7025 and https://github.com/docker/compose/pull/7046

**Comment by Juice10** [comment]
Bump

**Comment by glours** [comment]
@Juice10 did you recently test a build from the v2 branch? This issue has been addressed by #9386

**Comment by Juice10** [comment]
Bump!
